run_bin_test.go

```diff
@@ -396,6 +397,140 @@ func TestCoverageCollector_writeArgs(t *testing.T) {
 	}
 }
 
+func TestPreExec(t *testing.T) {
+	type args struct {
+		binPath      string
+		mainTestName string
+		env          []string
+		args         []string
+	}
+	var (
+		buffer1, buffer2 string
+	)
+
+	tests := []struct {
+		name        string
+		preCmdFuncs []PreCmdFunc
+		expected    string
+		args        args
+		wantErr     bool
+		wantErrMsg  string
+		buffer      *string
+	}{
+		{
+			name: "test running one prefunc",
+			args: args{
+				binPath:      "./test_bins/read_stdin.sh",
+				mainTestName: "TestRunMain",
+			},
+			buffer:      &buffer1,
+			preCmdFuncs: []PreCmdFunc{printToBufferPreFunc(&buffer1)},
+			expected:    "Hello from prefunc\n",
+		},
+		{
+			name: "test running two prefuncs",
+			args: args{
+				binPath:      "./test_bins/read_stdin.sh",
+				mainTestName: "TestRunMain",
+			},
+			buffer:      &buffer2,
+			preCmdFuncs: []PreCmdFunc{printToBufferPreFunc(&buffer2), printToBufferPreFunc(&buffer2)},
+			expected:    "Hello from prefunc\nHello from prefunc\n",
+		},
+		{
+			name: "test error in prefuncs",
+			args: args{
+				binPath:      "./test_bins/read_stdin.sh",
+				mainTestName: "TestRunMain",
+			},
+			preCmdFuncs: []PreCmdFunc{errPreCmdFunc()},
+			wantErr:     true,
+			wantErrMsg:  "oh no!",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := NewCoverageCollector("", false)
+			_ = c.Setup()
+			_, _, err := c.RunBinary(tt.args.binPath, tt.args.mainTestName, tt.args.env, tt.args.args, PreExec(tt.preCmdFuncs...))
+			if tt.wantErr {
+				require.Error(t, err)
+				require.Equal(t, tt.wantErrMsg, err.Error())
+			} else {
+				require.Equal(t, tt.expected, *tt.buffer)
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestPostExec(t *testing.T) {
+	type args struct {
+		binPath      string
+		mainTestName string
+		env          []string
+		args         []string
+	}
+	var (
+		buffer1, buffer2 string
+	)
+
+	tests := []struct {
+		name         string
+		postCmdFuncs []PostCmdFunc
+		expected     string
+		args         args
+		wantErr      bool
+		wantErrMsg   string
+		buffer       *string
+	}{
+		{
+			name: "test running one postfunc",
```

Comment:
Same question, is this test case needed?

Reply:
removed